Connection/Connection.php

```diff
@@ -90,6 +97,9 @@ public function getMailbox(): Mailbox
     
     public function testConnection(bool $throwExceptions = false): bool
     {
+        if(!$this->isEnabled()) {
+            throw new ConnectionException(array('Mailbox is not enabled'));
```

Comment:
testConnection should always return true or false and should not throw error in any case. You should return false here.